oss_src/unity/python/sframe/data_structures/sarray.py

```diff
@@ -3040,6 +3040,701 @@ def sort(self, ascending=True):
+        2    0.666667
+        3    0.666667
+        4    0.666667
+        dtype: float64
+
+        A rolling variance with a size of 3, centered around the current:
+        >>> sa.rolling_var(-1,1)
+        dtype: float
+        Rows: 5
+        [None, 0.6666666666666666, 0.6666666666666666, 0.6666666666666666, None] 
+
+        Pandas equivalent:
+        >>> pandas.rolling_var(series, 3, center=True)
+        0         NaN
+        1    0.666667
+        2    0.666667
+        3    0.666667
+        4         NaN
+        dtype: float64
+
+        A rolling variance with a window including the current and the 2 entries
+        following:
+        >>> sa.rolling_var(0,2)
+        dtype: float
+        Rows: 5
+        [0.6666666666666666, 0.6666666666666666, 0.6666666666666666, None, None]
+
+        A rolling variance with a window including the previous 2 entries NOT
+        including the current:
+        >>> sa.rolling_var(-2,-1)
+        dtype: float
+        Rows: 5
+        [None, None, 0.25, 0.25, 0.25]
+        """
+        min_observations = self.__check_min_observations(min_observations)
+        agg_op = '__builtin__var__'
+        return SArray(_proxy=self.__proxy__.builtin_rolling_apply(agg_op, window_start, window_end, min_observations))
+
+    def rolling_stdv(self, window_start, window_end, min_observations=None):
+        """
+        Calculate a new SArray of the standard deviation of different subsets
+        over this SArray.
+        
+        The subset that the standard deviation is calculated over is defined as
+        an inclusive range relative to the position to each value in the
+        SArray, using `window_start` and `window_end`. For a better
+        understanding of this, see the examples below.
+
+        Parameters
+        ----------
+        window_start : int
+            The start of the subset to calculate the standard deviation
+            relative to the current value. 
+
+        window_end : int
+            The end of the subset to calculate the standard deviation relative
+            to the current value. Must be greater than `window_start`.
+        
+        min_observations : int 
+            Minimum number of non-missing observations in window required to
+            calculate the standard deviation (otherwise result is None). None
+            signifies that the entire window must not include a missing value.
+            A negative number throws an error.
+
+        Returns
+        -------
+        out : SArray
+
+        Examples
+        --------
+        >>> import pandas
+        >>> sa = SArray([1,2,3,4,5]) 
+        >>> series = pandas.Series([1,2,3,4,5])
+
+        A rolling standard deviation with a window including the previous 2
+        entries including the current:
+        >>> sa.rolling_stdv(-2,0)
+        dtype: float
+        Rows: 5
+        [None, None, 0.816496580927726, 0.816496580927726, 0.816496580927726]
+
+        Pandas equivalent:
```

Comment:
It's a fair point, but I don't think that confusion will be crippling. My thought was that most people using us are going to be used to using time series stuff in pandas since it has such good support for it and is well established, and no one really finds the translator quickly. The model that we're using for specifying the rolling window, while I think it's better than pandas, it could be difficult to figure out at first without these if you're used to specifying the absolute size of the window.